.github/workflows/main.yml

```
@@ -0,0 +1,52 @@
+name: CI
+on: 
+  pull_request:
+  push:
+    branches:
+      - main
+      - add_GHA_based_CI
+
+jobs:
+  make_sdist:
+    name: Make SDist
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+
+      - run: python -m pip install build
+
+      - run: python -m build .
```

```suggestion
      - run: python -m build --sdist
```

No need to make an wheel here.